src/components/control_flow.rs

```diff
@@ -45,9 +94,59 @@ impl<I: IntoIterator<Item = Box<dyn Component<P>>>, P: Problem> From<I> for Box<
     }
 }
 
-/// Executes the child component as long as the condition is true.
+/// Loops the `body` while a `condition` is `true`.
+///
+/// This is equivalent to a `while` loop:
+/// ```no_run
+/// # fn condition() -> bool { true }
+/// # fn body() {}
+/// while condition() {
+///     body()
+/// }
+/// ```
 ///
-/// This corresponds to a `while` loop.
+/// # State
+///
+/// This component inserts and updates the current number of [`Iterations`].
```

Comment:
Maybe we should put a disclaimer here that we can, of course, also terminate depending on the number of evaluations, as this is more common.